mxresolv/mxresolv.go

```diff
@@ -98,20 +98,20 @@ func Lookup(ctx context.Context, hostname string) (retMxHosts []string, retImpli
 		return mxRecords[i].Pref < mxRecords[j].Pref ||
 			(mxRecords[i].Pref == mxRecords[j].Pref && mxRecords[i].Host < mxRecords[j].Host)
 	})
-	mxHosts := shuffleMXRecords(mxRecords)
+	mxHosts := ShuffleMXRecords(mxRecords)
 	if len(mxHosts) == 0 {
 		return cacheAndReturn(hostname, nil, nil, false, errNoValidMXHosts)
 	}
 	return cacheAndReturn(hostname, mxHosts, mxRecords, false, nil)
 }
 
-func shuffleMXRecords(mxRecords []*net.MX) []string {
+func ShuffleMXRecords(mxRecords []*net.MX) []string {
```

Comment:
Why did you make it a public function?

Reply:
Because I don't like writing tests for private methods. 😮‍💨 But in this case, I don't wanna do the correct thing and embed a DNS server we can query to retrieve proper MX records for testing. Instead we use publicly hosted MX records which don't have the specific regression configuration I wanted to target. So I made the method public and added tests for it.

I understand your position on this so I reverted and made the method private. 👍